core-bundle/src/Resources/contao/library/Contao/InsertTags.php

```diff
@@ -764,8 +767,58 @@ protected function doReplace($strBuffer, $blnCache)
 					break;
 
 				// Images
+				case 'figure':
+					// Expected format: <from>, <size>[, <configuration>[, <template>]]
+					$args = \array_slice($elements, 1);
+
+					if (\count($args) < 2)
+					{
+						$arrCache[$strTag] = '';
+
+						break;
+					}
+
+					// Allow implicit arrays for <size> and <configuration>
+					foreach (array(1, 2) as $index)
+					{
+						if (\array_key_exists($index, $args))
+						{
+							$arrayString = trim($args[$index]);
+
+							// Allow omitting outer brackets
+							if (0 !== strpos($arrayString, '['))
+							{
+								$arrayString = "[$arrayString]";
+							}
+
+							try
+							{
+								$args[$index] = ArrayString::parse($arrayString);
+							}
+							catch (ArrayStringParserException $e)
+							{
+								// Fallback to unprocessed string
+							}
+						}
+					}
+
+					$figureRenderer = $container->get(FigureRendererRuntime::class);
+
+					try
+					{
+						$arrCache[$strTag] = $figureRenderer->render(...$args);
+					}
+					catch (\Throwable $e)
+					{
+						$arrCache[$strTag] = '';
+					}
+
+					break;
+
 				case 'image':
 				case 'picture':
+					trigger_deprecation('contao/core-bundle', '4.11', 'Using the "image" and "picture" insert tags has been deprecated and will no longer work in Contao 5.0. Use the "figure" insert tag instead.');
```

Comment:
Are we sure we want to deprecate image and picture insert tags? What if I want to include an image without the `<figure>` image container?

Reply:
That's a valid point for the `image` insert tag, I guess. Although you could as well use an alternative template that simply outputs the image instead.